fairworkflows/rdf_wrapper.py

```diff
@@ -75,6 +75,18 @@ def remove_attribute(self, predicate, object=None):
         """
         self._rdf.remove((self.self_ref, predicate, object))
 
+    def shacl_validate(self, shaclfile="./shacl/plex-shapes.ttl"):
```

Comment:
Some questions:

- Why is this not a part of `validate()` method? e.g. just another reason why it is invalid.
- Why is this the same for Workflow and Step? I would expect the RDF to look different? (But I do not know much about shacl).
-

Reply:
* Yes, I initially wasn't sure about including it in ```validate()```, mainly since the error messages from shacl are not exactly user-friendly - at least, to our users, since our library supposedly abstracts out the need to worry about the underlying RDF graph. But I suppose it does belong there - maybe as the final pass? Just note that the user will receive error messages about shapes with long hash-like names that will mean nothing to them about why the validation failed. On the other hand, that would be an indication that our own, more user-friendly, validation has let something through that it shouldn't have.

* The shacl shapes file doesn't care whether all the given piece of RDF is all together (i.e. contains the whole workflow and explicit step descriptions) or if only a single step is given etc. If no RDF triples say stuff is a workflow then the workflow-specific contraints are not evaluated, so you can indeed use the same shacl file for everything (if that makes sense?)

Reply:
- If it doesn't produce user-friendly error messages we should not show them to the user. Indeed it could be a final pass check. And then the error message can just be: "RDF did not meet requirements" (we don't show the shacl messages).
- Ok, thanks for the clarification!

Reply:
As the shacl is more for us (currently) and the exact shapes file is still under consideration, I've made it so users only get shacl validation output if they ask for it. For now I think it might just confuse users to be presented either with tons of mad shacl errors (that they can do nothing about) or vague statements that the rdf isn't valid (which they also can't address). We can revisit this later once it's a bit more mature maybe...

Reply:
Alright sounds good!